src/unistd.rs

```diff
@@ -756,12 +756,27 @@ pub fn close(fd: RawFd) -> Result<()> {
     Errno::result(res).map(drop)
 }
 
+/// reads from a raw file descriptor.
+/// reads at most as many bytes as will fit into the provided buffer
+/// 
+/// returns the number of bytes read or an error
```

Comment:
The Error is pretty implied by the return value, I'd suggest instead "Returns the number of bytes successfully read". Also needs a period and capitalized first letter.